src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -244,8 +271,11 @@ public WorkItem Dequeue()
 
 
                 // Dequeue our work item
-                WorkItem work;
-                while (!_innerQueue.TryDequeue(out work)) { };
+                WorkItem work = null;
+                while (work == null)
+                    foreach (var q in _innerQueues)
+                        if (q.TryDequeue(out work))
+                            break;
```

Comment:
As I said, the code in question bothers me, but as you say it seems to be functionally equivalent. I am going to chalk it up to my feeling uncomfortable with the existing lockless implementation. You are correct that the two implementations are equivalent, but I think the second is more obvious, thus my disquiet.